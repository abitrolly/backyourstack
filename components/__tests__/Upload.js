import React from 'react';
import renderer from 'react-test-renderer';

import Upload from '../Upload';

test('Upload should render', () => {
  const component = renderer.create(<Upload />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Upload should render with feedbackPosition', () => {
  const component = renderer.create(<Upload feedbackPosition="inside" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Upload should render with style', () => {
  const component = renderer.create(
    <Upload style={{ border: '1px solid red' }} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
