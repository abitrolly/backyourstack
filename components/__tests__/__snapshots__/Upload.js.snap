// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload should render 1`] = `
<div
  className="jsx-201268674 uploadWrapper"
>
  <div
    aria-disabled={false}
    className="dropZoneComponent"
    onClick={[Function]}
    onDragEnter={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      className="jsx-201268674 text"
    >
      <p
        className="jsx-201268674"
      >
        Simply drag and drop files
        <br
          className="jsx-201268674"
        />
        or click to select files to upload.
      </p>
    </div>
    <input
      autoComplete="off"
      disabled={false}
      multiple={true}
      onChange={[Function]}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "opacity": 0.00001,
          "pointerEvents": "none",
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      type="file"
    />
  </div>
  <div
    className="jsx-201268674 uploadFeedback float"
  >
    <p
      className="jsx-201268674"
    >
      There was an error while uploading your files. At the moment, we support 
      <em>
        composer.json
      </em>
      , 
      <em>
        *.csproj
      </em>
      , 
      <em>
        packages.config
      </em>
      , 
      <em>
        Gopkg.lock
      </em>
      , 
      <em>
        Gemfile.lock
      </em>
       and 
      <em>
        requirements.txt
      </em>
      . Please try again. If the problem persists, please contact us.
    </p>
  </div>
</div>
`;

exports[`Upload should render with feedbackPosition 1`] = `
<div
  className="jsx-201268674 uploadWrapper"
>
  <div
    aria-disabled={false}
    className="dropZoneComponent"
    onClick={[Function]}
    onDragEnter={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      className="jsx-201268674 text"
    >
      <p
        className="jsx-201268674"
      >
        Simply drag and drop files
        <br
          className="jsx-201268674"
        />
        or click to select files to upload.
      </p>
    </div>
    <input
      autoComplete="off"
      disabled={false}
      multiple={true}
      onChange={[Function]}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "opacity": 0.00001,
          "pointerEvents": "none",
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      type="file"
    />
  </div>
  <div
    className="jsx-201268674 uploadFeedback inside"
  >
    <p
      className="jsx-201268674"
    >
      There was an error while uploading your files. At the moment, we support 
      <em>
        composer.json
      </em>
      , 
      <em>
        *.csproj
      </em>
      , 
      <em>
        packages.config
      </em>
      , 
      <em>
        Gopkg.lock
      </em>
      , 
      <em>
        Gemfile.lock
      </em>
       and 
      <em>
        requirements.txt
      </em>
      . Please try again. If the problem persists, please contact us.
    </p>
  </div>
</div>
`;

exports[`Upload should render with style 1`] = `
<div
  className="jsx-201268674 uploadWrapper"
>
  <div
    aria-disabled={false}
    className="dropZoneComponent"
    onClick={[Function]}
    onDragEnter={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      className="jsx-201268674 text"
    >
      <p
        className="jsx-201268674"
      >
        Simply drag and drop files
        <br
          className="jsx-201268674"
        />
        or click to select files to upload.
      </p>
    </div>
    <input
      autoComplete="off"
      disabled={false}
      multiple={true}
      onChange={[Function]}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "opacity": 0.00001,
          "pointerEvents": "none",
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      type="file"
    />
  </div>
  <div
    className="jsx-201268674 uploadFeedback float"
  >
    <p
      className="jsx-201268674"
    >
      There was an error while uploading your files. At the moment, we support 
      <em>
        composer.json
      </em>
      , 
      <em>
        *.csproj
      </em>
      , 
      <em>
        packages.config
      </em>
      , 
      <em>
        Gopkg.lock
      </em>
      , 
      <em>
        Gemfile.lock
      </em>
       and 
      <em>
        requirements.txt
      </em>
      . Please try again. If the problem persists, please contact us.
    </p>
  </div>
</div>
`;
